File: mentees/JayKawa/algorithms/searching/binary_search.c
#include <stdio.h>

int compare(const void* a, const void* b) 
{
    return (*(int*)a - *(int*)b);
}

int main()
{
    int n, f, low, high, mid,found;
    int a[100];

    printf("Enter the number of elements you want to store:\n");
    scanf("%d", &n);
    
    for(int i = 0; i < n; i++)
    {
        printf("Enter the %d element:\n", i + 1);
        scanf("%d", &a[i]);
    }

    printf("Enter the element you want to find:\n");
    scanf("%d", &f);

    qsort(a, n, sizeof(int), compare); 
    low = 0;
    high = n - 1;
    found=0;

    while(low <= high)
    {
        mid = (low + high) / 2;

        if(a[mid] == f)
        {
            break;
        }
        else if(f < a[mid])
        {
            high = mid - 1;
        }
        else if(f > a[mid])
        {
            low = mid + 1; 
        }
    }

    if(!found)
    {
        printf("Element not found.");
    }

    return 0;
}

File: mentees/JayKawa/algorithms/searching/Binary_search.java

import java.util.*;

public class Binary_search {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter the number of elements you want to store:");
        int n = sc.nextInt();
        int[] a = new int[n];

        for (int i = 0; i < n; i++) 
        {
            System.out.println("Enter the " + (i + 1) + " element:");
            a[i] = sc.nextInt();
        }

        System.out.println("Enter the element you want to find:");
        int f = sc.nextInt();

        Arrays.sort(a);

        int low = 0, high = n - 1;
        int mid=-1;
        boolean found = false;

        while (low <= high) 
        {
            mid = (low + high) / 2;

            if (a[mid] == f) 
            {
                found = true;
                break;
            } 
            else if (f < a[mid]) 
            {
                high = mid - 1;
            } 
            else 
            {
                low = mid + 1;
            }
        }

        if (found) 
        {
            System.out.println("Element found at position " + mid + ".");
        } 
        else 
        {
            System.out.println("Element not found.");
        }
        sc.close();
    }
}
